import_file/import_from_dump.go

```diff
@@ -0,0 +1,128 @@
+package import_file
+
+import (
+	"bufio"
+	"fmt"
+	spanneraccessor "github.com/GoogleCloudPlatform/spanner-migration-tool/accessors/spanner"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/common/constants"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/expressions_api"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/internal"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/logger"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/sources/common"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/sources/mysql"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/sources/postgres"
+	"go.uber.org/zap"
+	"os"
+)
+
+type ImportFromDump interface {
+	CreateSchema(dialect string) (*internal.Conv, error)
+	ImportData(conv *internal.Conv) error
+	Finalize() error
+}
+
+type ImportFromDumpImpl struct {
+	ProjectId       string
+	InstanceId      string
+	DatabaseName    string
+	DumpUri         string
+	dumpReader      *os.File
+	SourceFormat    string
+	SpannerAccessor spanneraccessor.SpannerAccessor
+	schemaToSpanner common.SchemaToSpannerInterface
+	dbDumpProcessor common.DbDump
+}
+
+func NewImportFromDump(
+	projectId string,
+	instanceId string,
+	databaseName string,
+	dumpUri string,
+	sourceFormat string,
+	spannerAccessor spanneraccessor.SpannerAccessor) (ImportFromDump, error) {
+	dbDump, err := getDbDump(sourceFormat)
+	if err != nil {
+		return nil, err
+	}
+	// TODO: handle GCS
+	dumpReader, err := os.Open(dumpUri)
+	if err != nil {
+		if err != nil {
+			return nil, fmt.Errorf(fmt.Sprintf("can't read dump file: %s due to: %v", dumpUri, err))
+		}
+	}
+
+	schemaToSpanner := &common.SchemaToSpannerImpl{
+		ExpressionVerificationAccessor: &expressions_api.ExpressionVerificationAccessorImpl{SpannerAccessor: spannerAccessor},
```

Comment:
Will the`mysqldump` flow call the expression verification API? If yes, this substantially increase the time required for schema conversion?

Reply:
Using expression verification API may not be viable for the kind of UX we want to provide for import flows. 

For context, this API makes calls to an actual spanner instance to verify the expressions that are part of the MySQL schema. For a small instance, something like a 100 processing unit instance (the free ones) this can take prohibitively long.

Since the goal of the import CLI is lightweight experimentation, I expect customers to be using such instances more so than migration, where a large instance is expected (and hence the above trade-off was acceptable).

FYI: @rohitwali

Reply:
The final goal is to remove the verifier but I am not making this part of this PR as it will require changes to SchemaToSpannerImpl, This will be done is a following PR.

Reply:
I have updated the PR to include this.